src/figuresClasses.ts

```diff
@@ -1,19 +1,78 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape = 'triangle';
+
+  sides: number[];
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('Invalid triangle sides');
+    }
 
+    this.color = color;
+    this.sides = [a, b, c];
+  }
+
+  getArea(): number {
+    const s = (this.sides[0] + this.sides[1] + this.sides[2]) / 2;
```

Comment:
Let's make your code cleaner 
```suggestion
    const s = (this.a + this.b + this.c) / 2;
```